src/ucp/core/ucp_context.c

```diff
@@ -438,14 +487,127 @@ void ucp_config_release(ucp_config_t *config)
 ucs_status_t ucp_config_modify(ucp_config_t *config, const char *name,
                                const char *value)
 {
-    return ucs_config_parser_set_value(config, ucp_config_table, name, value);
+    ucs_config_cached_key_t *key_val;
+    ucs_status_t ucs_status;
+
+    ucs_status = ucs_config_parser_set_value(config, ucp_config_table, name,
+                                             value);
+    if (ucs_status == UCS_OK) {
+        return UCS_OK;
+    }
+
+    ucs_status = ucp_cached_key_alloc(&key_val, name, value);
+    if (ucs_status == UCS_OK) {
```

Comment:
if (status != UCS_OK) {
 return status;
}

ucs_list_add_tail(...)
return UCS_OK;

Reply:
@yosefe After the UCT configurations have been cached here, it still needs return UCS_ERR_NO_ELEM. Or, some of the gtest cases can't pass. The reason is that some gtest cases use environment to change UCT configurations.

Reply:
can we update the gtests? it's weird that function returns failure but it actually saved the key
what if this function will be used in user code that checks for return value?

Reply:
@yosefe
It seems not easy to do that.
1. When calling ```uct_config_modify(void *config, const char *name, const char *value)```, it needs to know the ```config parameter```. However, there's no such ```argument``` there in gtest, so it use environment variable to change the configurations.
2. They key/valuel pair are not audited here, so it returns "UCS_ERR_NO_ELEM".

One method is to check the UCT configurations key/value in ucp_config_modify, than ucp_config_modify could return "UCS_ERR_NO_ELEM" if there's really no such element. However, we've argeed that there's no need to audit the key/value previously.

Reply:
but you are already doing this check when printing the configuration.. so maybe do it here?
another option is that gtest will not use the return value from ucp_config_modify()

Reply:
I'll try to check the key/value in ```ucp_config_modify```

Reply:
can you do this in next PR? and in this PR, do not check, and print as-is?

Reply:
Could I keep the return value of ```ucp_config_modify()``` to be ```UCS_ERR_NO_ELEM``` for the cached UCT configurations? If not, I'll try to modify gtest.

Reply:
let's see how complicated it is to modify gtest

Reply:
I've modified gtest to avoid calling ```ucp_config_modify``` directly.